community-operators/planetscale/planetscale-operator.v0.1.7.clusterserviceversion.yaml

```diff
@@ -0,0 +1,273 @@
+  version: 0.1.7
+  maturity: alpha
+  maintainers:
+  - name: PlanetScale
+    email: contact@planetscale.com
+  links:
+  - name: PlanetScale
+    url: https://planetscale.com
+  provider:
+    name: PlanetScale
+  labels:
+    name: planetscale-operator
+  selector:
+    matchLabels:
+      name: planetscale-operator
+  installModes:
+  - type: OwnNamespace
+    supported: true
+  - type: SingleNamespace
+    supported: true
+  - type: MultiNamespace
+    supported: false
+  - type: AllNamespaces
+    supported: false
+  install:
+    strategy: deployment
+    spec:
+      clusterPermissions:
+      - serviceAccountName: planetscale-operator
+        rules:
+        - apiGroups:
+          - security.openshift.io
+          resources:
+          - securitycontextconstraints
+          verbs:
+          - use
+          resourceNames:
+          - anyuid
+      - serviceAccountName: default
```

Comment:
Why does the default SA need this additional clusterRole?

Reply:
Both the operator pod and the application pods run as a specific user and do not support OpenShift's restricted SCC, so I had to add the `anyuid` rule to both service accounts.

I'm not totally clear on the implications and interactions of the `use` permission, so if there's a better way, I'm happy to test and change this.

Reply:
I'm not concerned about the SCC itself as much as I am that the `default` service account is being given heightened permissions. OLM will deploy multiple operators to the namespace with access to the `default` service account so any of these operators now have the SCC.

Reply:
yeah we should just make our own SA (and give it the SCC like the other ones). OLM should do this for you if you just change the name from default to anything. Make sure the name matches in the Deployment and the RBAC.

Reply:
Oh I guess the Operator's SA is not the one that will run the end Deployment or StatefulSet

Reply:
Right, the operator creates pods that use the `default` service account, so we're kind of stuck. Working on a CSV description update that documents this behavior and what users should do about it.

Reply:
Thanks @joelanford, could you please make sure to create a bug for the operator?